pkg/core/blockchain.go

```diff
@@ -399,6 +407,54 @@ func (bc *Blockchain) init() error {
 	return bc.updateExtensibleWhitelist(bHeight)
 }
 
+// JumpToState is an atomic operation that changes Blockchain state to the one
+// specified by the state sync point p. All the data needed for the jump must be
+// collected by the state sync module.
+func (bc *Blockchain) JumpToState(p uint32) error {
+	bc.lock.Lock()
+	bc.log.Info("jumping to state sync point", zap.Uint32("state sync point", p))
+
+	block, err := bc.dao.GetBlock(bc.headerHashes[p])
+	if err != nil {
+		bc.lock.Unlock()
+		return fmt.Errorf("failed to get current block: %w", err)
+	}
+	err = bc.dao.StoreAsCurrentBlock(block, nil)
+	if err != nil {
+		bc.lock.Unlock()
+		return fmt.Errorf("failed to store current block: %w", err)
+	}
+	bc.blockHeight = p
+	bc.persistedHeight = p
+
```

Comment:
Fixed for `topBlock`, `extensible` is already there, but moved it under mutex protection.

Why do we need to set `defaultBlockWitness` here? It will be updated during regular `ApplyPolicyToTxSet` functioning, won't it?